core/app/models/spree/fulfilment_changer.rb

```diff
@@ -141,11 +143,27 @@ def handle_stock_counts?
       current_shipment.order.completed? && current_stock_location != desired_stock_location
     end
 
-    def default_on_hand_quantity
+    def get_available_quantity
+      positive_stock = ->(shipment) do
```

Comment:
Did you think about moving this method to StockLocation or Shipment?

Reply:
I'm not sure it would be worth it, I think there are no other occurrences in the codebase where we need the positive `on_hand` quantity, so I'm a bit hesitant about adding it to an ActiveRecord model that's not fully related to the concept, but of course I'm open to different opinions.

Alternatively, I'm thinking we may be using `Spree::Stock::Quantifier` and add a new method there, such as `#positive_stock`... this seems a more appropriate context to me. The stock quantifier object is less efficient in this context, but I'm thinking this should not be a real problem since this service should not see intensive usage, and after all, we're already using the [Stock Quantifier in this class](https://github.com/solidusio/solidus/blob/main/core/app/models/spree/fulfilment_changer.rb#L159)... what's your take on this?

Reply:
@kennyadsl I've moved that lambda into the stock quantifier object, for now I've left the changes separated from the previous version, you can take a look at them starting from [this commit](https://github.com/solidusio/solidus/pull/5670/commits/5a21d930d0cdae69da4b291896d49597b6d6ab8b), let me know what you think, thanks!